pkgs/os-specific/linux/dcgm/default.nix

```diff
@@ -116,14 +108,18 @@ in gcc11Stdenv.mkDerivation rec {
   ];
 
   buildInputs = [
-    plog.dev # header-only
-    tclap_1_4 # header-only
-
+    # Header-only
     catch2
-    fmt_9
-    jsoncpp-static
-    libevent-nossl-static
-    yaml-cpp
+    plog.dev
+    tclap_1_4
+
+    # Dependencies that can be either static or dynamic.
+    (fmt_9.override { enableShared = !static; }) # DCGM's build uses the static outputs regardless of enableShared
+    (yaml-cpp.override { inherit static; stdenv = gcc11Stdenv; })
+
+    # TODO: Dependencies that DCGM's CMake hard-codes to be static-only.
+    (jsoncpp.override { enableStatic = true; })
+    (libevent.override { sslSupport = false; static = true; })
```

Comment:
Ok there's actually one more concern I didn't raise before in relation to the static/dynamic packages and overrides: I think it's desirable to avoid ad hoc overrides (e.g. overrides in let-in expressions or as here) and directly consume the top-level attributes instead. This way it's easier to override dependencies and to control them from overlays. For instance, instead of an ad hoc `libevent-nossl-static` in a `let-in` we could have dedicated an attribute with the same name in `pkgs`.

I'd also check if the build works with all overrides removed. I'd test `pkgs.dcgm` and `pkgsStatic.dcgm`.

By avoiding all this overspecification in `dcgm/default.nix` we keep the package more configurable/overridable. That said, I've no idea what the original motivation behind all of these flags was, so you probably know better

Reply:
DCGM's CMake isn't nearly flexible enough to support that kind of arbitrary overriding. If we wanted to support that, we'd have a *lot* more work to do (namely, patching DCGM's build system in non-trivial ways). I do not have the bandwidth to find or maintain such a patchset.